### PR TITLE
Update the dockwidget plugin to have three tabs: home, catalog and doc api loading externally

### PR DESCRIPTION
**Pull request by XavierCLL**
Update UI dockwidget using three tabs; home plugin, GEE catalog and GEE doc api. The GEE catalog and GEE doc api load the original webpage in mobile version. I propose to load the catalogue and doc api externally:

* **Reasons:**
The original way was so complicated, and that is not the idea of this plugin, we need work on creating more function emulated from EE API

* **Current problems:**
The GEE catalog and GEE doc api are so heavy web pages even in the mobile version, I will report it to Google developers to fix that. Currently, the navigation in those pages is slow inside the plugin dockwidget, and the links don't work appropriately. Search not implemented yet (waiting if the previous problems to be fixed)

### CODE REVIEW

**Comment by XavierCLL** [comment]
Hello Gena, I need to push more and more changes, then I will push this PP and if you want to check this after go ahead. I think for not delay the development, I will commit to the repo directly. Regards

File: ee_plugin/Makefile [hunk 1]
```diff
@@ -23,37 +23,47 @@ PY_FILES = \
...
+# QGISDIR points to the location where your plugin should be installed.
+# This varies by platform, relative to your HOME directory:
+#	* Linux:
+#	  .local/share/QGIS/QGIS3/profiles/default
+#	* Mac OS X:
+#	  Library/Application Support/QGIS/QGIS3/profiles/default
+#	* Windows:
+#	  AppData\Roaming\QGIS\QGIS3\profiles\default'
+
+QGISDIR=.local/share/QGIS/QGIS3/profiles/default
```
Comment (gena) [inline ee_plugin/Makefile]: This .local will give issues on Windows?

File: ee_plugin/utils.py [hunk 2]
```diff
@@ -60,25 +60,25 @@ def get_layer_by_name(name):
...
     
-        update_ee_image_layer(image, layer, visibility, opacity)
+        update_ee_image_layer(image, layer, shown, opacity)
     else:
-        add_ee_image_layer(image, name, visibility, opacity)
+        add_ee_image_layer(image, name, shown, opacity)
 
 
-def add_ee_catalog_image(name, asset_name, vis_props, collection_props):
+def add_ee_catalog_image(name, asset_name, visParams, collection_props):
```
Comment (gena) [inline ee_plugin/utils.py]: ... camel notation is not PEP8 enough

**Comment by gena** [comment]
My QGIS hangs :)

**Comment by gena** [comment]
I'd suggest placing Home and API to the external page (e.g. GitHub / more python-ish docs), there is no reason to embed it into the QGIS plugin. As for the catalog - we will have to make some indexed page to speed-up loading everything. And use paging support to show items, avoiding loading everything at once.

File: ee_plugin/utils.py (hunk 2 above)
Comment (XavierCLL) [reply ee_plugin/utils.py]: yes of course I know it is not PEP8 rule but I try to follow the params name used by the addLayer function (to maintain consistency): https://developers.google.com/earth-engine/api_docs#map.addlayer

**Comment by XavierCLL** [comment]
yes, you're right, I was thinking the same thing, then:

1) I will separate the "home" tab to convert it as documentation

2) while we found a solution to embed the api and doc tab, I will change it as external links, are you agree?

File: ee_plugin/Makefile (hunk 1 above)
Comment (XavierCLL) [reply ee_plugin/Makefile]: mmm this requires some OS conditional for set this variable...

**Comment by gena** [comment]
One way could be to put it into a menu, e.g. Plugins > Google Earth Engine > User Guide, About and then link to for example plugin Wiki page on GitHub, it will be easier to provide examples there as well.

**Pull request merged**